src/lib.rs

```diff
@@ -1298,6 +1298,62 @@ impl<T: Clone> Receiver<T> {
             listener: None,
         }
     }
+
+    /// A low level poll function that returns items of the same format handed back by
+    /// [`Receiver::recv()`] and [`Receiver::recv_direct()`]. This can be useful for building
+    /// stream implementations which use a [`Receiver`] under the hood and want to know if the
+    /// stream has overflowed.
+    ///
+    /// Prefer to use [`Receiver::recv()`] or [`Receiver::recv_direct()`] when otherwise possible.
+    pub fn poll_recv(
+        mut self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+    ) -> Poll<Option<Result<T, RecvError>>> {
```

Comment:
RecvError::Closed is not possible here so would be nice to change the error type to `struct Overflow(pub n)` but this works as well.

```suggestion
    ) -> Poll<Option<Result<T, Overflow>>> {
```

Reply:
Yeahh I wondered about this too. @zeenix what do you reckon? 

1. Stick with `RecvError` from `poll_recv()` even though we'll only use the Overflow variant
2. Add `OverflowError` struct or similar and return `Result<T, OverflowError>`
3. Return  overflow count directly ie `Result<T, u64>` and document this.

I'm leaning to 2 but happy with whatever you prefer :)

I also added a quick test for `poll_recv()` for my own sanity check as much as anything!

Reply:
> Stick with returning `Result<T,RecvError>` from `poll_recv()` even though we'll only use the Overflow variant

Yes, I think this is the best way forward. It's more future-proof this way. 

This reminds me that it would be good to document possible errors from this method in a `Errors` section (as per the conventions).

Reply:
Actually looks like `RecvError` is non_exhaustive so any other variants will require API break anyway. So I'd say we go with @niklasad1's suggestion.